gmakefile.test

```diff
@@ -263,7 +267,7 @@ else ifdef searchin
 else ifdef argsearch
   TESTTARGETS := $(foreach v,$(alltesttargets),$(if $(findstring $(argsearch),$($(v)_ARGS)),$(v)))
 else ifdef globsearch
-  TESTTARGETS := $(shell $(PYTHON) -c"import sys,fnmatch,itertools; m=[fnmatch.filter(sys.argv[2].split(),p) for p in sys.argv[1].split()]; print(' '.join(list(itertools.chain.from_iterable(m))))" '$(globsearch)' '$(alltesttargets)')
+  TESTTARGETS := $(shell $(PYTHON) $(PETSCCONFIGDIR)/query_tests.py 'name' '$(globsearch)')
```

Comment:
This won't work with PETSc 3.13.0.
On the other hand, it is always possible to use search in our case.

Reply:
@jkruzik I reverted this but added a comment. I also added a few more minor changes. Please look again. Please test it on your end.

Note it didn't work for me even with master. I don't wanna spend even more time by figuring out what's the problem. Let's look at it again at some point in the future.

In my ideal world, PETSc test harness should improve support for external packages like PERMON or SLEPc so that we don't have to maintain this whole file but include some PETSc one instead.